packages/evolution-common/src/types/Optional.types.ts

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2024, Polytechnique Montreal and contributors
+ *
+ * This file is licensed under the MIT License.
+ * License text available at https://opensource.org/licenses/MIT
+ */
+
+// These are collection of optional types, that can mimic Rust Option<T> type
+
+/**
+ * Optional boolean
+ *
+ * @export
+ * @typedef {OptionalBoolean}
+ */
+export type OptionalBoolean = boolean | undefined;
```

Comment:
This could be a single generic type `Optional<T> = T | undefined`, to avoid having the multiplication of distinct concrete types.

Reply:
done!